code/feature_extraction/FeatureExtractor.py

```diff
@@ -37,7 +37,6 @@ def extract_features(self):
 
         # Get targets
         labels = self.__get_targets(self.df['truthClass'])
-        self.df.drop('truthClass', axis=1, inplace=True)
```

Comment:
You can still drop the truthClass here, as all the classification steps are already assuming that the dataframe only consists of features

Reply:
I know! But when you're executing the `extract_features` method twice, the second call raises an error because it can't drop a column that has already been dropped.